crypto/dkg_feldmanvss.go

```diff
@@ -331,8 +344,13 @@ func (s *feldmanVSSstate) receiveVerifVector(origin index, data []byte) {
 	s.vA = make([]pointG2, s.threshold+1)
 	err := readVerifVector(s.vA, data)
 	if err != nil {
-		s.processor.FlagMisbehavior(int(origin),
-			fmt.Sprintf("reading the verification vector failed: %s", err))
+		if IsInvalidInputsError(err) { // case where vector format is invalid
+			s.vAReceived = true
+			s.validKey = false
+			s.processor.Disqualify(int(origin),
+				fmt.Sprintf("reading the verification vector failed: %s", err))
+		}
+		// verification vector should not be tagged as received if unexpected error
```

Comment:
To align with error handling best practices, we should explicitly handle the generic unexpected error case, by crashing the component or propagating it to the caller.

I think the reasonable way to do this would be to return an error from `receiveVerifVector` and propagate it back to `HandleBroadcastMsg` for unexpected error cases. We would want to make a similar change for the other `receive***` functions as well. Probably out of scope for this PR, but maybe we could add some TODOs or tech debt issues to update error handling in the future?

Reply:
oh good catch, this definitely needs to be changed. The unexpected error case is actually not possible in those cases, I'll clean it up.

Reply:
cleaned up in https://github.com/onflow/flow-go/pull/2970/commits/e0841b6ce8e216756a42594b93e83e41d879ed18. 
Btw, Flow DKG does not use that portion of code, it's a lighter version of DKG, and we "overload" its `HandkeBroadcastMsg` function.